src/test/java/com/epam/jamp2/service/impl/CalculationCommandExecutionServiceImplTest.java

```diff
@@ -0,0 +1,95 @@
+package com.epam.jamp2.service.impl;
+
+import com.epam.jamp2.model.CalculationCommand;
+import com.epam.jamp2.model.CommandFormatException;
+import com.epam.jamp2.model.UnknownCurrencyException;
+import com.epam.jamp2.model.Value;
+import com.epam.jamp2.service.CalculationCommandExecutionService;
+import com.epam.jamp2.service.FxRatesService;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Optional;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CalculationCommandExecutionServiceImplTest {
+  private static CalculationCommandExecutionService calculationService;
+
+  @Mock
+  FxRatesService mockedFXRatesService;
+
+  @InjectMocks
+  private CalculationCommandExecutionService mockedCalculationService = new CalculationCommandExecutionServiceImpl();
+
+  @BeforeClass
+  public static void setup() {
+    calculationService = new CalculationCommandExecutionServiceImpl();
+  }
+
+  @AfterClass
+  public static void teardown() {
+    calculationService = null;
+  }
+
+  @Test
+  public void shouldCalculateCorrectWithUSDCurrency() {
```

Comment:
I would generalize this test a bit, since USD is just one of plenty possible currencies and it doesn't really make any difference if you test your logic againt USD or CNY for example. I would implement "shouldCalculateForCorrectFormatWithoutConversion"